### PR TITLE
Enabling RSpec IndexedLet Rubocop rule

### PR DESCRIPTION
**Pull request by GarryHurleyJr**
https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecindexedlet

### What are you trying to accomplish?
The RSpec IndexedLet rubocop rule has been disabled.  This PR will re-enable it on the whole ecosystem.

### Anything you want to highlight for special attention from reviewers?

This was essentially a renaming of variables in the rspec files.  It shoud not affect anything.

### How will you know you've accomplished your goal?

This cop just asks us to rename variables.  The tests should run as normal.

### Checklist

- [x] I have run the complete test suite to ensure all tests and linters pass.
- [x] I have thoroughly tested my code changes to ensure they work as expected, including adding additional tests for new functionality.
- [x] I have written clear and descriptive commit messages.
- [x] I have provided a detailed description of the changes in the pull request, including the problem it addresses, how it fixes the problem, and any relevant details about the implementation.
- [x] I have ensured that the code is well-documented and easy to understand.

### CODE REVIEW

File: bundler/spec/dependabot/bundler/helper_spec.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@
     LOCKFILE
   end
 
-  let(:lockfile_bundled_with_v1) do
+  let(:first_version_lockfile) do
```
Comment (jurre) [inline bundler/spec/dependabot/bundler/helper_spec.rb:17]: v1 refers to the version of Bundler here. Not it being the first occurrence of the variable.

**Review by jurre — CHANGES_REQUESTED** [review]
Swapping out `file1` for `first_file` seems not in the spirit of the rule. 

> It makes reading the test harder because it’s not clear what exactly is tested by this particular example.

That is not addressed by changing things in this way. 

There are also a bunch of scenarios here where `v1` or `v2` refers to the version of the package manager. We should configure a pattern to allow that, because that is very clear and the current changes actually make those cases much harder to understand, so we'd end up in a worse situation there.

File: bundler/spec/dependabot/bundler/helper_spec.rb (hunk 1 above)
Comment (GarryHurleyJr) [reply bundler/spec/dependabot/bundler/helper_spec.rb:17]: I can change that one to lockfile_bundle_with_version_one.  It keeps the same spirit of that original name.  In another fie I changed v6 to version_six for that reason.

Comment (GarryHurleyJr) [reply bundler/spec/dependabot/bundler/helper_spec.rb]: corrected with commit 4c83ce7.  Please review

**Comment by jurre** [comment]
None of these changes seem like they are an improvement, the spirit of this linter rule is to make the tests more readable. All of the changes I see here are along the lines of changing `file1` to `file_one`, and while that passes the linter, it does not seem like a meaningful improvement.

If we can't think of descriptive names for these, I'd rather we disable the rule than make these changes.

**Comment by GarryHurleyJr** [comment]
Instead of changing all of these files, per discussions, we are going to simply disable this rule as it does not gain us anything in the long run.

**Pull request closed without merging**